### DISCUSSION TITLE
Kitty is stretching fonts vertically

### DISCUSSION THREAD

**Discussion by charbelnicolas** [kovidgoyal/kitty#5387] [Q&A]
Is there a reason why kitty is stretching the fonts vertically? If you compare the two screenshots below and watch the video you will notice Xterm font rendering looks much nicer and proportional.

Xterm:
![20220816_121609](https://user-images.githubusercontent.com/14960414/184944170-26aa1643-496b-4d30-830a-77046346ea44.png)

Kitty:
![20220816_121625](https://user-images.githubusercontent.com/14960414/184944176-8a4d5239-9e99-4b23-b051-851eaf9b51c4.png)
 

https://user-images.githubusercontent.com/14960414/184944614-d7ba8a8d-9f20-4657-bb7d-06a14aeadc96.mp4

Any help is appreciated, I can't enjoy reading text on kitty anymore after using Xterm :(

I would like to know if there is a fix for it, thank you.

**Comment by page-down** [comment]
Because using the same font does not mean that it will be rendered in exactly the same way in different software.
And I don't think kitty stretched the font.

From your two screenshots, I can see that the cell width pixels are the same and the height differs by one pixel.

- In different software, the final calculated cell width is equal, does not mean that the actual font size is equal.
- The same unit name and the same value does not mean that you can get exactly the same rendered size.
- Different software may not calculate the width of the cell the same, some allow the characters to overlap, some require the cell to fit perfectly, so the margins may be different.
- When the font size is small, one pixel will greatly affect the size when applying font anti-aliasing or sub-pixel rendering.

In this case, I don't think your font sizes are the same, it looks like there is a difference in fractions.
(Again, just because you've configured the same integer font size doesn't mean you'll get the same final render size, between different rendering engines.)

> I would like to know if there is a fix for it ...

There is no "fix" for it.
You can try to adjust the font size or modify the source code to fine-tune it, but it is difficult to get exactly the same rendering result as another software.

**Comment by charbelnicolas** [comment]
Alright, fair enough, I still think Kitty could improve its font rendering though.

Thanks for taking the time to answer!
